cpp_guidelines/guidelines.md

```diff
@@ -0,0 +1,496 @@
+  * `.clang-format`: `BitFieldColonSpacing: Both`
+
+**Example:**
+```c++
+float abcdef = 42.f;
+int xyz      = 3;
+double klmn  = 10.0;
+
+int aaaaaaa : 42;
+int bbbb    : 21;
+int cc      : 10;
+
+#define SHORT                  10
+#define VERY_VERY_LONG         21
+#define THIS_IS_ABSURDELY_LONG 42
+#define evil_func(x)           (x + x)
+#define another_evil_func(x)   (x * x)
+
+#define FOO                             \
+    int aaaaaa;                         \
+    int bbbb;                           \
+    int cc;                             \
+```
+
+### Align operators, colons and commas on the left hand side.
+
+**Reason:** For operators this follows the syntax you would use by writing by
+hand. The commas and colons are kept here for consistency.
+
+**Enforced by:**
+  * `.clang-format`: `AlignOperands: true`
+  * `.clang-format`: `BreakBeforeBinaryOperators: All`
+  * `.clang-format`: `BreakBeforeTernaryOperators: true`
+  * `.clang-format`: `BreakConstructorInitializers: BeforeComma`
+  * `.clang-format`: `BreakInheritanceList: BeforeComma`
+  * `.clang-format`: `OperandAlignmentStyle: Align`
+
+**Example:**
+```c++
+int aaa = bbbbbbbbbb
+        + cccccccccc;
+
+class foo
+    : public first_parent
+    , public second_parent
+{
+    foo()
+        : first_member{some_value}
+        , second_member{another_value}
+};
+```
+
+### Align trailing comments.
+
+**Reason:** This makes the documentation easier to read.
+
+**Enforced by:**
+  * `.clang-format`: `AlignTrailingComments: true`
+
+**Example:**
+```c++
+int aaaaaaaaa; // A comment
+int bbb;       // Another comment.
+```
+
+### Continued lines are indented.
+
+**Reason:** This helps to visually separate them from the surrounding scope.
+
+**Enforced by:**
+  * `.clang-format`: `ContinuationIndentWidth: 4`
+
+**Example:**
+```c++
+auto foo = some_long_initializer{value1,
+    value2, {inner_value_1,
+        inner_value_2}};
+```
+
+### Case labels are not indented.
+
+**Reason**: Case labels do not open a new scope. Indentation helps to
+differentiate between scopes.
+
+**Enforced by:**
+  * `.clang-format`: `IndentCaseLabels: false`
+
+**Example:**
+```c++
+switch(statement)
+{
+case first_case:
+    /* something */
+    break;
+
+case second_case:
+    /* something else */
+    break;
+}
+```
+
+### Case blocks are indented.
+
+**Reason**: Case blocks are part of a scope (opened by `switch`) and should be
+indented accordingly.
+
+**Enforced by:**
+  * `.clang-format`: `IndentCaseBlocks: true`
+
+```c++
+switch(statement)
+{
+case first_case:
+    /* something */
+    break;
+
+case second_case:
+    /* something else */
+    break;
+}
+```
+
+### `extern "C"` blocks are indented.
+
+**Reason:** `extern "C"` opens a new scope (with special semantics). It
+should be indented accordingly.
+
+**Enforced by:**
+  * `.clang-format`: `IndentExternBlock: AfterExternBlock`
+
+### `goto` labels are not indented.
+
+***DO NOT USE `goto`!***
```

Comment:
I would turn that into: Use goto sparingly. Or just reference the C++ core guidelines: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-goto